main.go

```diff
@@ -514,6 +509,41 @@ func main() {
 	}
 }
 
+func registerClient(ctx context.Context, conf *config.Config, logger *zap.SugaredLogger) *k8s.AdmissionWebhookRegisterClient {
+	var registerClient = k8s.AdmissionWebhookRegisterClient{
+		Logger: logger.Named("admissionWebhookRegisterClient"),
+	}
+
+	err := registerClient.Register(ctx, conf)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+
+	return &registerClient
+}
+
+func getX509Source(ctx context.Context, logger *zap.SugaredLogger) *workloadapi.X509Source {
+	logger.Infof("Obtaining X509 Certificate Source")
+	source, err := workloadapi.NewX509Source(ctx)
+	if err != nil {
+		logger.Fatalf("error getting x509 source: %+v", err)
+		return nil
+	}
+	return source
+}
+
+func prepareTLSConfig(conf *config.Config, source *workloadapi.X509Source) *tls.Config {
+	tlsConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+	}
+	if strings.EqualFold(conf.WebhookMode, config.ModeAuto) {
+		tlsConfig.Certificates = append([]tls.Certificate(nil), conf.GetOrResolveCertificate())
+	} else {
```

Comment:
I think it would be better to parse both Auto and Manual modes, and return an error if user typed an invalid mode.

Reply:
Done